queue/queue.go

```diff
@@ -323,6 +342,51 @@ func (q *Queue) Len() int64 {
 	return int64(len(q.items))
 }
 
+// GetItems returns items in this queue.
+func (q *Queue) GetItems() []interface{} {
+	q.lock.Lock()
+	defer q.lock.Unlock()
+
+	return q.items
+}
+
+// Search takes a function and returns a list of items that
+// match the checker.  This does not wait and remove items.
+func (q *Queue) Search(checker func(item interface{}) bool) ([]interface{}) {
+	if checker == nil {
+		return nil
+	}
+
+	q.lock.Lock()
```

Comment:
I realize that this is an ancient PR, but it might be better to do a single deferred unlock here as is done elsewhere in this file.